src/cpp/rtps/reader/RTPSReader.cpp

```diff
@@ -376,6 +376,20 @@ bool RTPSReader::is_sample_valid(
     return true;
 }
 
+bool RTPSReader::add_statistics_listener(
+            std::shared_ptr<fastdds::statistics::IListener> listener)
+{
+    (void)listener;
+    return false;
+}
+
+bool RTPSReader::remove_statistics_listener(
+            std::shared_ptr<fastdds::statistics::IListener> listener)
+{
+     (void)listener;
+     return false;
+}
+
```

Comment:
You need to include the `#ifdef FASTDDS_STATISTICS` compilation guard in order to build the code without this CMake option enabled.